extide/gradle/src/org/netbeans/modules/gradle/cache/ProjectInfoDiskCache.java

```diff
@@ -83,7 +86,7 @@ public QualifiedProjectInfo(Quality quality, NbProjectInfo pinfo) {
 
         @Override
         public Map<String, Object> getExt() {
-            return Collections.emptyMap();
+            return ext != null ? ext : Collections.emptyMap();
```

Comment:
Actually this is the primary fix. The transient data (I called that ext for whatever reason) shall be not cached to the disk, but when NetBeans process the project information it holds important data on binary, source and javadoc location which are stored in the GradleAtrifactStore. Returning an empty map when the project is not restored from serialization, made that these location data were never stored in the artifact store.